lib/rules/lint-simple-unless.js

```diff
@@ -1,14 +1,71 @@
 'use strict';
 
 const Rule = require('./base');
+const createErrorMessage = require('../helpers/create-error-message');
 
 const messages = {
   followingElseBlock: 'Using an {{else}} block with {{unless}} should be avoided.',
   asElseUnlessBlock: 'Using an `{{else unless}}` block should be avoided.',
   withHelper: 'Using {{unless}} in combination with other helpers should be avoided.'
 };
 
+const DEFAULT_CONFIG = {
+  whitelist: [],
+  maxHelpers: 0
+};
+
+function isValidConfigObjectFormat(config) {
+  for (let key in config) {
+    let value = config[key];
+    let valueIsArray = Array.isArray(value);
+
+    if (key === 'whitelist' && !valueIsArray) {
+      return false;
+    } else if (DEFAULT_CONFIG[key] === undefined){
+      return false;
+    }
+  }
+
+  return true;
+}
+
 module.exports = class LintSimpleUnless extends Rule {
+  parseConfig(config) {
+    let configType = typeof config;
+
+    switch (configType) {
+    case 'boolean':
+      // if `true` use `DEFAULT_CONFIG`
+      return config ? DEFAULT_CONFIG : false;
+    case 'number':
+      return {
+        maxHelpers: config
+      };
+    case 'object':
+      if (Array.isArray(config)) {
+        return {
+          whitelist: config,
+          maxHelpers: 1
+        };
+      } else if (isValidConfigObjectFormat(config)) {
+        // default any missing keys to empty values
+        return {
+          whitelist: config.whitelist || [],
+          maxHelpers: config.maxHelpers || 1
```

Comment:
Why would this default to 1?

Reply:
when somebody uses and object without maxHelpers, instead of an array.
e.g. :
```js
'simple-unless': { whitelist: ['not', 'or'] }
```